studio/actions/ConfirmPublishWithi18n.ts

```diff
@@ -130,10 +134,29 @@ export const ConfirmPublishWithi18nAction = ({ type, id, onComplete }: IResolver
       throw new Error(UiMessages.errors.baseDocumentNotPublished)
     } else {
       // Add firstPublishedField
-      if (document && requiresFirstPublished.includes(type) && !document?.[FIRST_PUBLISHED_AT_FIELD_NAME]) {
+      // https://github.com/sanity-io/sanity/issues/2179
+      const revisions = await fetch(
+        `https://${projectId}.api.sanity.io/v${apiVersion}/data/history/${dataset}/transactions/${document?._id}?excludeContent=true`,
+        {
+          method: 'GET',
+          headers: new Headers({
+            Authorization: `Bearer ${token}`,
+          }),
+        },
+      )
+        .then((res) => res.text())
+        .catch((err: Error) => console.error(err))
+
+      const hasBeenPublished = revisions?.includes(`"_id":"${document?._id.replace('drafts.', '')}"`)
```

Comment:
Seems weird that this is the way to check if a document has been published before or not, but at the same time it doesn't surprise me.

Reply:
I couldn't find a better way to do this. I will ask in the sanity channel, but I do agree this is super ugly.